src/amp-story-player/amp-story-component-manager.js

```diff
@@ -43,19 +55,39 @@ export class AmpStoryComponentManager {
       return;
     }
 
-    const intersectingCallback = (entries) => {
+    const ioPrerenderCb = (entries) => {
+      entries.forEach((entry) => {
+        if (!entry.isIntersecting) {
+          return;
+        }
+        elImpl.prerenderCallback();
+
+        prerenderObserver.unobserve(elImpl.getElement());
+      });
+    };
+
+    const rootMarginInPercentage =
+      (MIN_VIEWPORT_PRERENDER_DISTANCE * 100).toString() + '%';
+    const prerenderObserver = new IntersectionObserver(ioPrerenderCb, {
+      rootMargin: rootMarginInPercentage,
+    });
+    prerenderObserver.observe(elImpl.getElement());
+
+    const ioLayoutCb = (entries) => {
       entries.forEach((entry) => {
         if (!entry.isIntersecting) {
           return;
         }
         elImpl.layoutCallback();
```

Comment:
Does intersectionObserver trigger more than once for an element? Like if you scroll back for a second time to the player, does it call `layoutCallback` again or just once?

Reply:
It would normally trigger whenever the criteria of the IO is met (i.e. whatever is described on its rootMargin or threshold in its init parameters). But we are also calling IO.unobserve() once it triggers the first time, which stops observing the element and thus won't trigger a second time.

We also have the if (this.isLaidOut_) return; in the player code to stop from running a second time in case the fallback for the IO is needed in unsupported browsers.

Reply:
👍